src/core/Akka/Actor/Stash/StashFactory.cs

```diff
@@ -50,15 +48,14 @@ public static IStash CreateStash(this IActorContext context, IActorStash actorIn
         /// <returns>TBD</returns>
         public static IStash CreateStash(this IActorContext context, Type actorType)
         {
-            if(actorType.Implements<IWithBoundedStash>())
-            {
+            if (actorType.Implements<IWithBoundedStash>())
                 return new BoundedStashImpl(context);
-            }
 
-            if(actorType.Implements<IWithUnboundedStash>())
-            {
+            if (actorType.Implements<IWithUnboundedStash>())
                 return new UnboundedStashImpl(context);
-            }
+
+            if (actorType.Implements<IWithUnrestrictedStash>())
+                return new UnrestrictedStashImpl(context);
 
             throw new ArgumentException($"Actor {actorType} implements an unrecognized subclass of {typeof(IActorStash)} - cannot instantiate", nameof(actorType));
         }
```

Comment:
`BoundedStashImpl`, `UnboundedStashImpl` and `UnrestrictedStashImpl` are all empty implementations, so we could get away with a single type (i.e., `StashSupportImpl`) like in the JVM.